src/consumption.cpp

```diff
@@ -1089,6 +976,126 @@ void Character::modify_morale( item &food, const int nutr )
         add_morale( MORALE_FOOD_GOOD, fun.first, fun.second, morale_time, morale_time / 2, false,
                     food.type );
     }
+
+    // Morale bonus for eating unspoiled food at a nearby table
+    if( !food.rotten() && !food.has_flag( flag_ALLERGEN_JUNK ) ) {
+        bool has_chair_nearby = false;
+        for( const tripoint &pt : g->m.points_in_radius( pos(), 1 ) ) {
```

Comment:
Since you have factored out `has_nearby_table`, having also `has_nearby_chair` makes perfect sense to me. That would be symmetrical.

The only thing that bothers me is they're not required be adjacent to one another. This still can be fine if we assume chairs are moveable (the character uses them and then puts them back). However, vehicle seats aren't the case.

Another approach could be finding a nearby chair (like in `cata::optional<tripoint> find_nearby_char(const tripoint&, int)`) and then finding a table adjacent to that chair (like in `find_nearby_table(chair_pos, 1)`).

Reply:
I agree, it makes more sense to require the table and chair to be adjacent to one another. But do we still require them both to be adjacent to the player? For example, do you think these would be OK?

```
..T  .C.
.C.  ..T
.@.  .@.
```

Or should all three be adjacent to one another, like this?

```
.TC  .T.
.@.  .@C
...  ...
```

Reply:
This may be a reasonable counterexample in support of the current behavior - I don't have a chair, but I have a Beetle, and I can set up a tourist table right next to my Beetle to eat there. So the idea is that I'm sitting in the driver's seat with the door open (or even just the window down, let's pretend), and my food is sitting on the table, like a tray at a drive-up:

![image](https://user-images.githubusercontent.com/238395/74089200-ff10ec80-4a5b-11ea-8d27-d665da4ae624.png)

Due to the game geometry, there's a full tile gap between them, but I think this is a legitimate case where the morale bonus should apply for having both chair and table, although they aren't directly adjacent. So maybe it's best to leave it as-is, and give players that little extra benefit?

Reply:
I guess the player needs to be adjacent to a chair (to sit on it), but not necessarily to a table. The chair, on the other hand, has to be adjacent to the table. So this is OK:

```
..T
.C.
.@.
```
I wasn't exactly correct speaking of nearby chairs. We have to examine all the nearby chairs, because "the closest" (anyone of them) can lack a table. Consider this layout:
```
.C.     TC.
..@     ..@
TC.     .C.
```

We can also become more permissive and allow a chair in the vicinity (say, two tiles away), but there must be a passable way to it:

```
...T
...C
.@..
....
```

Reply:
> Due to the game geometry, there's a full tile gap between them, but I think this is a legitimate case where the morale bonus should apply for having both chair and table, although they aren't directly adjacent. So maybe it's best to leave it as-is, and give players that little extra benefit?

Yeah, I didn't consider this case. Makes sense, but I have to think about it. Requiring the player to stand exactly between then (in the "doorway"), but never sitting on the seat in order to the bonus be applicable, doesn't strike me as a very intuitive game mechanic.

Reply:
> Requiring the player to stand exactly between then (in the "doorway"), but never sitting on the seat in order to get the bonus, doesn't strike me as a very intuitive game mechanic.

Yeah, it is kind of an edge case. I suppose another way is to park the car diagonally and place the table adjacent to the seat :-)

As you have pointed out, it gets more complicated with multiple chairs, with and without adjacent tables. I'm concerned that the more rules we have, the more the player will have to understand in order to get the morale bonus. The current rule of "1 adjacent table + 1 adjacent chair" is at least easy to understand.

Reply:
I'm fine with keeping the current behavior for now, but extracting a `has_nearby_chair` function still would be nice. I suppose a separate issue for the conundrum is also reasonable.